src/main/java/org/folio/innreach/domain/service/impl/InnReachTransactionServiceImpl.java

```diff
@@ -36,12 +93,68 @@ public InnReachResponseDTO createInnReachTransactionItemHold(String trackingId,
-    InnReachTransaction transaction = new InnReachTransaction();
-    transaction.setTrackingId(trackingId);
-    transaction.setCentralServerCode(centralCode);
-    transaction.setType(TransactionType.ITEM);
-    transaction.setState(InnReachTransaction.TransactionState.ITEM_HOLD);
-    return transaction;
+  @Override
+  public void createItemRequest(String trackingId) {
+    var transaction = repository.fetchOneByTrackingId(trackingId).orElseThrow(() ->
+      new EntityNotFoundException("Transaction with trackingId = " + trackingId + " not found.")
+    );
+    var itemId = transaction.getHold().getItemId();
+    String errorReason;
+
+    var item = inventoryClient.getItemById(itemId);
+    var requests = requestsClient.findRequests(itemId);
+
+    if (itemIsRequestable(item, requests)) {
+      try {
+        var comment = "INN-Reach request: Patron Agency - " + transaction.getHold().getPatronAgencyCode() +
+          ", Pickup Location - " + mapper.map(transaction.getHold().getPickupLocation());
+        var centralServerId = centralServerRepository.fetchOneByCentralCode(transaction.getCentralServerCode())
+          .orElseThrow(() ->
+            new EntityNotFoundException("Central server with central code = " + transaction.getCentralServerCode() + " not found.")
+          ).getId();
+        var patronType = ((TransactionItemHold) transaction.getHold()).getCentralPatronType();
+        var patronBarcode = centralPatronTypeMappingRepository.
+          findOneByCentralServerIdAndCentralPatronType(centralServerId, patronType);
+        var user = usersClient.query("(barcode==\"" + patronBarcode + "\")").getResult().get(0);
+        var defaultServicePointId = inventoryStorageClient.findServicePointsUsers(UUID.fromString(user.getId()))
+          .getServicePointsUsers().get(0).getDefaultServicePointId();
+        var requestExpirationDate = transaction.getHold().getNeedBefore() == null ? null :
+          OffsetDateTime.ofInstant(Instant.ofEpochMilli(transaction.getHold().getNeedBefore()), ZoneOffset.UTC);
+
+        var newRequest = RequestDTO.builder()
+          .requestType(item.getStatus() == AVAILABLE ? PAGE : HOLD)
+          .itemId(itemId)
+          .requesterId(UUID.fromString(user.getId()))
+          .pickupServicePointId(defaultServicePointId)
+          .requestExpirationDate(requestExpirationDate)
+          .patronComments(comment)
+          .requestDate(transaction.getCreatedDate())
+          .fulfilmentPreference(HOLD_SHELF)
+          .build();
+        var createdRequest = requestsClient.sendRequest(newRequest);
+
+        transaction.getHold().setFolioRequestId(createdRequest.getId());
+        transaction.getHold().setFolioPatronId(UUID.fromString(user.getId()));
+        transaction.getHold().setFolioItemId(itemId);
+        repository.save(transaction);
+      } catch (Exception e) {
+        errorReason = "Request not permitted";
+        //issue a "Owning site cancels request" call to the central server
```

Comment:
You've stated in PR description:

> "Owning site cancels request" is not sent when request creation failed, as this feature is yet to be implemented.

Is there any additional story for implementing this request?

My understanding it should be a call from mod-inn-reach to central server:

> Issue a "Owning site cancels request" call to the central server
>
> *  See D2IR API docs for specification of this request

thus it can be done quite easily with specific feing client.